R/site.R

```diff
@@ -68,3 +68,48 @@ rmd_needs_render <- function(path) {
 rmd_hash <- function(path) {
   digest::digest(path, file = TRUE, algo = "xxhash64")
 }
+
+hugodown_site <- function(input = ".", output_format = NULL) {
+  # read config and output directory
+  config <- hugo_config(input)
+  output_dir <- hugo_config_str(config, "publishdir")
+
+  list(
+    # name for site (used as default name for publishing)
+    name = hugo_config_str(config, "title"),
+
+    # output directory (e.g. "public")
+    output_dir = output_dir,
+
+    # render_site callback
+    render = function(input_file, output_format, envir, quiet, ...) {
+
+      # Render the entire site
+      if (is.null(input_file)) {
+        hugo_build(input)
```

Comment:
I think we should set `relativeUrls` here; I'll add a parameter to `hugo_build()` to make it easier.

Reply:
Yes I think that solves the deployment problem very nicely!

Reply:
Updated to do this. The directions for deploying to RSC should now just be:

```r
library(rmarkdown)
publish_site()
```

And the user will be prompted to select a server, connect their account, etc. URLs will be relative so will all work on RSC. This won't affect buggy themes that rely on absoluteURLs for things like Twitter cards b/c internal publishing doesn't rely on that sort of public metadata.

cc @apreshill @sloop

Reply:
And a buggy theme can always be fixed by setting the `baseURL` and re-publishing.